src/form.js

```diff
@@ -446,23 +389,12 @@ define([
             this.fireChange(mug);
         },
         /**
-         * Walk merged data and control trees calling valueFunc for each mug
-         *
-         * If common nodes in merged tree branches are not in the same order
-         * then the order of nodes in the control tree will be preserved, and
-         * all nodes in the data tree that are not also in the control tree will
-         * be visited last.
+         * Walk through the mugs of the tree and call valueFunc on each mug
          *
          * @param valueFunc - a function called for each mug in the tree. The
          * single argument is the mug.
          */
-        mergedTreeMap: function (valueFunc) {
-            function getID(node) {
-                return node.getID() || node.getValue().ufid;
-            }
-            function hasControl(node) {
-                return !node.getValue().options.isDataOnly;
-            }
+        walkMugs: function (valueFunc) {
```

Comment:
This got quite a bit simpler, which is nice. It should be possible to make it simpler yet with `Tree.walk`. This example has not been tested, so may need to be tweaked:

```js
walkMugs: function (valueFunc) {
    function callback(mug, nodeID, processChildren) {
        if (mug !== null) {
            valueFunc(mug);
        }
        processChildren();
    }
    this.tree.walk(callback);
}
```